xdsl/dialects/arith.py

```diff
@@ -225,8 +229,6 @@ def __hash__(self) -> int:
 class BinaryOperationWithFastMath(Generic[_T], BinaryOperation[_T]):
     fastmath = opt_prop_def(FastMathFlagsAttr)
 
-    traits = frozenset((Pure(),))
```

Comment:
Why this change?

Reply:
Well they are not all Pure, and if we start actually using those traits, this is a timebomb!

Pure is NoMemoryEffect+AlwaysSpeculatable;
AlwaysSpeculatable implies that an operation never has undefined behaviour,
which is not the case of e.g. divisions, where dividing by 0 is UB

Reply:
Ah nice, thats very clear